TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java

```diff
@@ -346,6 +351,21 @@ private void getTangoPermission(String permissionType) {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        mRestartTangoAlertReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                mTangoServiceClientNode.callTangoConnectService(TangoConnectRequest.DISCONNECT);
```

Comment:
This is not safe in case TangoDisconnect would take a long time. You need to work with the callbacks `onTangoDisconnect` or add a new option to the message `RECONNECT`.